converter/shieldhit/parser.py

```diff
@@ -26,6 +26,8 @@ def save_configs(self, target_dir: str):
         Save the configs as text files in the target_dir.
         The files are: beam.dat, mat.dat, detect.dat and geo.dat.
         """
+        if not path.exists(target_dir):
+            raise ValueError("Target directory does not exist.")
         target_dir = path.abspath(target_dir)
```

Comment:
It's very bad practice to modify parameters passed to a function.

```suggestion
        target_dir = path.abspath(target_dir)
```

I would simply use approach like that:

```
        for file_name, content in self.get_configs_json().items():
            with open(Path(target_dir, file_name), 'w') as conf_f:
                conf_f.write(content)
```

I believe using absolute dirs is not needed at all.